conan/cli/args.py

```diff
@@ -56,45 +58,76 @@ def add_common_install_arguments(parser):
+                     default=None,
+                     type=None,
+                     choices=None,
+                     required=False,
+                     help=None,
+                     metavar=None,
+                     contexts=None):
+            if nargs == 0:
+                raise ValueError('nargs for append actions must be != 0; if arg '
+                                 'strings are not supplying the value to append, '
+                                 'the append const action may be more appropriate')
+            if const is not None and nargs != argparse.OPTIONAL:
+                raise ValueError('nargs must be %r to supply const' % argparse.OPTIONAL)
+            super(ContextAllAction, self).__init__(
+                option_strings=option_strings,
+                dest=dest,
+                nargs=nargs,
+                const=const,
+                default=default,
+                type=type,
+                choices=choices,
+                required=required,
+                help=help,
+                metavar=metavar)
+            self.contexts = contexts
+
+        def __call__(self, action_parser, namespace, values, option_string=None):
+            for context in self.contexts:
+                items = getattr(namespace, self.dest + "_" + context, None)
+                items = items[:] if items else []
+                items.append(values)
+                setattr(namespace, self.dest + "_" + context, items)
+
+    def create_config(short, long, example=None):
+        parser.add_argument(f"-{short}", f"--{long}",
+                            default=None,
                             action="append",
-                            dest='conf_{}'.format(machine),
-                            help='Configuration to build the package, overwriting the defaults'
-                                 ' ({} machine). e.g.: -c{} '
-                                 'tools.cmake.cmaketoolchain:generator=Xcode'.format(machine,
-                                                                                     short_suffix))
-
-    for item_fn in [options_args, profile_args, settings_args, conf_args]:
-        item_fn("host", "",
-                "")  # By default it is the HOST, the one we are building binaries for
-        item_fn("build", ":b", ":build")
-        item_fn("host", ":h", ":host")
+                            dest=f"{long}_host",
+                            metavar=long.upper(),
+                            help='Apply the specified profile. '
+                                 f'By default, or if specifying -{short}:h (--{long}:host), it applies to the host context. '
+                                 f'Use -{short}:b (--{long}:build) to specify the build context, '
+                                 f'or -{short}:a (--{long}:all) to specify both contexts at once'
+                                  + ('' if not example else f". Example: {example}"))
+        contexts = ["build", "host"]
```

Comment:
If we hardcode this as a const, instead of an arg, probably the Action can be further simplified?